### DISCUSSION TITLE
GC Form Builder

### DISCUSSION THREAD

**Discussion by adam-code-labx** [lunarphp/lunar#1061] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
Building forms in GC is very time consuming. Currently all modifications needs to be done directly in the blade templates.

**Describe the solution you'd like**
Considering all the form inputs are already extracted to blade components we can extend this without any other form package dependencies. As a developer using both Filament and Nova defining the fields in a php class is a much better way to setup the form schema, and far more extendable. I have taken a similar approach to the abstract tables class and created a set of traits for the the form builder.

**Describe alternatives you've considered**
Obviously considered filament forms but rejected. I understand why as we have our own custom fields 👍 

**Additional context**
@glennjacobs @alecritson you said anything with filament forms will be rejected. What I have created here is based around the existing components. If accepted I will also implement the changes to #478 

I have also merged #474 refactored using the new method which as you can see this is much cleaner way to define a form.

![carbon](https://user-images.githubusercontent.com/53559175/184693525-2249729b-7595-41cc-8abb-9e9404791232.png)

**Comment by wychoong** [comment]
> I have also merged #474 refactored using the new method

I think this should be separated from the Brand pr to get it merged as this Form Builder definitely requires more discussion, and rework on the form later.

**Comment by adam-code-labx** [comment]
Sure no problem @wychoong 
Once this has been reviewed I can isolate this from the brands and table PRs however I believe they will get merged to develop soon so it’s up to @alecritson on how he wants to proceed 😜

**Comment by adam-code-labx** [comment]
Also to note I will create a set of tests around this feature.
Once approved 👍

**Comment by alecritson** [comment]
Thanks @adam-code-labx I've put comments on the PR :)
